lib/fetch/index.js

```diff
@@ -2002,7 +2003,7 @@ async function httpNetworkFetch (
                 location = val
               }
 
-              headers.append(key, val)
+              if (isValidHeaderName(key)) headers.append(key, val)
```

Comment:
are these headers expected to be ignored, or should an error be thrown? It doesn't make sense to double validate every header we receive.

Reply:
I too think double checking is useless.
If `:status` is passed as an example of an http2 header, the http2 test fails.
I will comment the solution later.

Reply:
Some solutions,
- Change core so that http2 headers, e.g. `:status`, are not passed.
- Use try-catch.
- Check only the leading colon.

Reply:
None of the specific `h2` headers should be passed as those are automatically appended upon request. Although it can be passed, they are going to be overwritten.
On Chrome, `fetch` throws when trying to append them as headers

Reply:
@metcoder95
Is it correct to identify pseudo-headers with a leading colon and not pass them to Headers#append?
Or change the core?

Reply:
Probably due to the :status pseudo-header.

Reply:
Probably due to the `:status` pseudo-header.

Reply:
The pseudo-header `:status` seems to be the cause

Reply:
please undo both of these changes, we can reevaluate in a dedicated issue

Reply:
Sure, changes will be undone.